### PR TITLE
JMS: Akka 2.6 features

### PR DESCRIPTION
**Pull request by dotbg**
References #2485.

### CODE REVIEW

**Review by seglo — COMMENTED** [review]
Thanks a lot for picking up some of these Akka 2.6 compatibility tasks :+1:

Looking good so far. Can you enable `fatalWarnings` in `build.sbt` to see if there are any other changes required?

```
lazy val jms = alpakkaProject("jms", "jms", Dependencies.Jms, fatalWarnings := true)
```

**Comment by dotbg** [comment]
will do today

File: jms/src/test/scala/akka/stream/alpakka/jms/JmsProducerRetrySpec.scala [hunk 1]
```diff
@@ -59,7 +62,10 @@ class JmsProducerRetrySpec extends JmsSpec {
       server.stop() // crash.
 
       Thread.sleep(1000)
-      server.restart() // recover.
+      server.stop()
+      // https://activemq.apache.org/how-do-i-restart-embedded-broker.html
+      server.service.waitUntilStopped()
+      server.start(true) // recover.
```
Comment (ennru) [inline jms/src/test/scala/akka/stream/alpakka/jms/JmsProducerRetrySpec.scala:68]: Thank you for modernising this, I believe one `server.stop()` can be removed.

**Review by ennru — APPROVED** [review]
LGTM, just one question.

File: jms/src/test/scala/akka/stream/alpakka/jms/JmsProducerRetrySpec.scala (hunk 1 above)
Comment (dotbg) [reply jms/src/test/scala/akka/stream/alpakka/jms/JmsProducerRetrySpec.scala:68]: I will try.

**Review by ennru — APPROVED** [review]
LGTM.

**Comment by ennru** [comment]
Thank you!

**Comment by dotbg** [comment]
Not at all.

**Pull request merged**